Python/optimizer_bytecodes.c

```diff
@@ -980,6 +984,23 @@ dummy_func(void) {
         }
     }
 
+    op(_GUARD_NOS_NULL, (null, unused -- null, unused)) {
+        if (sym_is_null(null)) {
+            REPLACE_OP(this_instr, _NOP, 0, 0);
+        }
+        sym_set_null(null);
+    }
+
+    op(_GUARD_CALLABLE_TYPE_1, (callable, unused, unused2 -- callable, unused, unused2)) {
+        if (sym_is_const(ctx, callable)) {
+            PyObject *value = sym_get_const(ctx, callable);
+            assert(value != NULL);
+            if (value == (PyObject *)&PyType_Type) {
+                REPLACE_OP(this_instr, _NOP, 0, 0);
+            }
+        }
```

Comment:
I *might* like this simpler spelling a bit more, but your call which you prefer.

```suggestion
        if (sym_get_const(ctx, callable) == (PyObject *)&PyType_Type) {
            REPLACE_OP(this_instr, _NOP, 0, 0);
        }
```